### PR TITLE
Site Selector: Switch to on-demand api version verification design like iOS

### LINKED ISSUE #691: Fetching api version for store always shows store has wrong api version
**Describe the bug**
If an error occurs while fetching the WooCommerce store version from the API, the **Store Selector** screen will display that store as not having the correct API version (even though it may). Here is a snippet of the original code:

```
@Suppress("unused")
@Subscribe(threadMode = ThreadMode.MAIN)
fun onApiVersionFetched(event: OnApiVersionFetched) {
    if (event.apiVersion == WooCommerceStore.WOO_API_NAMESPACE_V3) {
        supportedWCSites.add(event.site)
    } else {
        unsupportedWCSites.add(event.site)
    }

    val totalSitesChecked = supportedWCSites.size + unsupportedWCSites.size
    if (totalSitesChecked == wooCommerceStore.getWooCommerceSites().size) {
        loginEpilogueView?.showStoreList(supportedWCSites, unsupportedWCSites)
    }
}
```

if `event.isError == true`, the `event.apiVersion` will be set to an empty string, so `event.apiVersion != WooCommerceStore.WOO_API_NAMESPACE_V3` and therefore the store will show up as having the wrong api version. The fix will be to add some proper error handling.

### PR DESCRIPTION
**Pull request by AmandaRiu**
This PR fixes #691 and changes the design of the site selector to more closely match the iOS version.

The former design validated all stores up front and then displayed stores that did not meet the minimum api version requirements in a separate section. The problem was that if the store could not be validated for one reason or another, they would automatically show up in the "Update required" section, even though they _may_ be running the correct version of WooCommerce. A couple ideas were thrown around to address this like creating a separate section for the stores that could not be verified, but in the end we decided to match the iOS design since it allows more flexibility on how to handle the validation response at the individual site level.

The new design will show all Woo stores in a single list and then validation will happen when the user _**selects a store**_. There are three possible outcomes of this validation:
1. The store passes -> the **continue** button will enable
2. The store's api version does not meet the minimum requirements (failed) -> a dialog will be shown advising the user to upgrade their WooCommerce plugin.
3. The store cannot be verified for some reason (error) -> display a snack error message

### CODE REVIEW

**Comment by AmandaRiu** [comment]
@nbradbury here is the PR I was telling you about. Everything should work, just needs some additional testing and maybe some refinement on the verbiage used. Feel free to modify as you see fit.

**Comment by nbradbury** [comment]
@AmandaRiu Code looks good and works just fine, but I found it odd to do the verification as soon as a site is tapped (plus the screen just looked odd with the disabled Continue button and no site selected). To me it made more sense to do the verification when the Continue button is tapped, so I made that change.

I also tightened up the wording a bit, as we both agreed it was too wordy.

screenshot_1547850425

**Comment by AmandaRiu** [comment]
Created new PR using a new branch created specifically for hotfix: https://github.com/woocommerce/woocommerce-android/pull/695
